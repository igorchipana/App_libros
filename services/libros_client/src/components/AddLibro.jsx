import React from 'react';

const stylesboton = {
    
     float:"right"
};

const styles1 = {
    
     marginTop:"0px"
};
 
const styles2 = {
    
     background:"#D3D3D3",
     marginTop:"30px",
     padding:"50px"
}; 
const styles3 = {
    
     marginTop:"10px"
};





const AddLibro = (props) => {
  return (
   <div class="col-md-12">
                     <div class="well well-sm">
                   <form className="form-horizontal" style={styles2} onSubmit={(event) => props.addLibro(event)}>
                    
                            <fieldset>
                              <legend class="text-center header">Registro de libros</legend>
                            <div className="col-md-12">
                                <span className="col-md-1 col-md-offset-2 text-center"><i className="fa fa-user bigicon"></i></span>
                                <div>
                                    <input id="producto" id="nombre" name="nombre" type="text" placeholder="Nombre del libro" className="form-control" require value={props.nombre} onChange={props.handleChange}></input>
                                </div>
                            </div>
                            <div className="form-group" className="col-md-12" style={styles3}>
                                <span className="col-md-1 col-md-offset-2 text-center"><i className="fa fa-user bigicon"></i></span>
                                <div>
                                    <input  id="autor" name="autor" type="text" placeholder="Autor del libro" className="form-control" require value={props.autor}  onChange={props.handleChange}></input>
                                </div>
                            </div>
                            <div className="form-group" className="col-md-12" style={styles3}>
                                <span className="col-md-1 col-md-offset-2 text-center"><i className="fa fa-user bigicon"></i></span>
                                <div >
                                    <input id="costo" name="costo" type="number" placeholder="Precio del libro" className="form-control" require value={props.costo} onChange={props.handleChange}></input>
                                </div>
                            </div>

    
                            <div className="form-group" className="col-md-12" style={styles3}>
                                <span className="col-md-1 col-md-offset-2 text-center"><i className="fa fa-pencil-square-o bigicon"></i></span>
                                <div>
                                    <textarea className="form-control" id="descripcion" name="descripcion" placeholder="Descripcion del libro" rows="7" require value={props.descripcion} onChange={props.handleChange}></textarea>
                                </div>
                            </div>

                            <div  className="form-group" className="col-md-12" style={styles3} >
                            <h4  type="text" >Seleccione Categoria:</h4>
                            </div>

                            <div className="form-group" className="col-md-12"  style={styles1} >
                        
                                <select className="col-md-4" name="categoria"  id="categoria" placeholder="Descripcion del producto"  require value={props.categoria} onChange={props.handleChange}>
                                <option values="option1">Terror</option>
                                <option values="option2">Realismo</option>
                                <option values="option3">Drama</option>
                                <option values="option4">Familia</option>
                                <option values="option5">Educaci√≥n</option>
                                <option values="option6">Ficcion</option>
                                <option values="option7">Caricatura</option>
                              </select>
                            </div>

                           <div className="form-group" style={stylesboton}>
                                <div class="col-md-12" >
                                    <button type="submit" class="btn btn-primary btn-lg " id="hola">Registrar</button>
                                </div>
                            </div>
                        </fieldset>
                     
                    </form>
                       </div>
                        </div>
               
  )
};

export default AddLibro;