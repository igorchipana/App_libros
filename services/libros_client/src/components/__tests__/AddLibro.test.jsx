import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import AddLibro from '../AddLibro';

test('AddLibro renders properly', () => {
  const wrapper = shallow(<AddLibro/>);
  const element = wrapper.find('form');
  expect(element.find('input').length).toBe(3);
  expect(element.find('input').get(0).props.name).toBe('nombre');
  expect(element.find('input').get(1).props.name).toBe('autor');
  expect(element.find('input').get(2).props.name).toBe('costo');
  expect(element.find('textarea').length).toBe(1);
  expect(element.find('textarea').get(0).props.name).toBe('descripcion');
   expect(element.find('select').length).toBe(1);
  expect(element.find('select').get(0).props.name).toBe('categoria');
   expect(element.find('button').length).toBe(1);
  expect(element.find('button').get(0).props.type).toBe('submit');
});

test('AddLibro renders a snapshot properly', () => {
  const tree = renderer.create(<AddLibro/>).toJSON();
  expect(tree).toMatchSnapshot();
});