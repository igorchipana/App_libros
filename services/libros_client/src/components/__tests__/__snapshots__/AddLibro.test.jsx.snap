// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddLibro renders a snapshot properly 1`] = `
<div
  class="col-md-12"
>
  <div
    class="well well-sm"
  >
    <form
      className="form-horizontal"
      onSubmit={[Function]}
      style={
        Object {
          "background": "#D3D3D3",
          "marginTop": "30px",
          "padding": "50px",
        }
      }
    >
      <fieldset>
        <legend
          class="text-center header"
        >
          Registro de libros
        </legend>
        <div
          className="col-md-12"
        >
          <span
            className="col-md-1 col-md-offset-2 text-center"
          >
            <i
              className="fa fa-user bigicon"
            />
          </span>
          <div>
            <input
              className="form-control"
              id="nombre"
              name="nombre"
              placeholder="Nombre del libro"
              require={true}
              type="text"
            />
          </div>
        </div>
        <div
          className="col-md-12"
          style={
            Object {
              "marginTop": "10px",
            }
          }
        >
          <span
            className="col-md-1 col-md-offset-2 text-center"
          >
            <i
              className="fa fa-user bigicon"
            />
          </span>
          <div>
            <input
              className="form-control"
              id="autor"
              name="autor"
              placeholder="Autor del libro"
              require={true}
              type="text"
            />
          </div>
        </div>
        <div
          className="col-md-12"
          style={
            Object {
              "marginTop": "10px",
            }
          }
        >
          <span
            className="col-md-1 col-md-offset-2 text-center"
          >
            <i
              className="fa fa-user bigicon"
            />
          </span>
          <div>
            <input
              className="form-control"
              id="costo"
              name="costo"
              placeholder="Precio del libro"
              require={true}
              type="number"
            />
          </div>
        </div>
        <div
          className="col-md-12"
          style={
            Object {
              "marginTop": "10px",
            }
          }
        >
          <span
            className="col-md-1 col-md-offset-2 text-center"
          >
            <i
              className="fa fa-pencil-square-o bigicon"
            />
          </span>
          <div>
            <textarea
              className="form-control"
              id="descripcion"
              name="descripcion"
              placeholder="Descripcion del libro"
              require={true}
              rows="7"
            />
          </div>
        </div>
        <div
          className="col-md-12"
          style={
            Object {
              "marginTop": "10px",
            }
          }
        >
          <h4
            type="text"
          >
            Seleccione Categoria:
          </h4>
        </div>
        <div
          className="col-md-12"
          style={
            Object {
              "marginTop": "0px",
            }
          }
        >
          <select
            className="col-md-4"
            id="categoria"
            name="categoria"
            placeholder="Descripcion del producto"
            require={true}
          >
            <option
              values="option1"
            >
              Terror
            </option>
            <option
              values="option2"
            >
              Realismo
            </option>
            <option
              values="option3"
            >
              Drama
            </option>
            <option
              values="option4"
            >
              Familia
            </option>
            <option
              values="option5"
            >
              Educaci√≥n
            </option>
            <option
              values="option6"
            >
              Ficcion
            </option>
            <option
              values="option7"
            >
              Caricatura
            </option>
          </select>
        </div>
        <div
          className="form-group"
          style={
            Object {
              "float": "right",
            }
          }
        >
          <div
            class="col-md-12"
          >
            <button
              class="btn btn-primary btn-lg "
              id="hola"
              type="submit"
            >
              Registrar
            </button>
          </div>
        </div>
      </fieldset>
    </form>
  </div>
</div>
`;
